src/cache_invalidate.c

```diff
@@ -142,7 +129,7 @@ cache_invalidate_xact_end(XactEvent event, void *arg)
 			 * change since the transaction hasn't been committed and other
 			 * backends cannot have the invalid state.
 			 */
-			cache_invalidate_all();
+			cache_invalidate_relcache_all();
```

Comment:
One question is what happens when you, in one transaction, add a foreign server and open a connection to it (thus adding it to the cache), followed by an abort of the transaction. Will the connection remain in the cache or will it receive a syscache invalidation event?

Reply:
Might be good to have isolation test for this

Reply:
This situation is not possible, since we are not allow to execute `add_data_node()` as part of a multi-statement transaction.